src/codegen/function.rs

```diff
@@ -456,3 +456,36 @@ pub fn body_chunk_futures(
 
     Ok(body)
 }
+
+pub(super) fn generate_static_to_str(
+    w: &mut dyn Write,
+    env: &Env,
+    function: &analysis::functions::Info,
+) -> Result<()> {
+    writeln!(w)?;
+    version_condition(w, env, function.version, false, 1)?;
+
+    let vis = match function.visibility {
+        Visibility::Public => "pub ",
+        _ => "",
+    };
+
+    writeln!(w, "\t{}fn {}(self) -> &'static str {{", vis, function.name)?;
+
+    // TODO: Can we somehow reuse function_body_chunk::Builder here?
```

Comment:
If you feel bored, probably. We ideally want to get rid of all that code generation abstraction as it makes things more complicated than necessary (thus also the experiment with quote)

Reply:
Not really, unfortunately.

`quote` might be a good starting point to refactor certain chunks of the code. Not going to be me though, I'd love to introduce it in smaller places to show how it simplifies things, but a full-on refactor is something for someone more invested in `gir` ;)

Reply:
If you feel like it, submit a PR with some minor parts moved over to quote after this big one here is in :)

Reply:
That has always been the plan since removing it... But I converted parts of this PR that have massively changed since then :grimacing: